site/src/pages/UserSettingsPage/NotificationsPage/NotificationsPage.tsx

```diff
@@ -60,6 +62,41 @@ export const NotificationsPage: FC = () => {
 	const updatePreferences = useMutation(
 		updateUserNotificationPreferences(user.id, queryClient),
 	);
+	const [searchParams] = useSearchParams();
+	const templateId = searchParams.get("disabled");
+
+	useEffect(() => {
+		if (templateId && templatesByGroup.isSuccess && templatesByGroup.data) {
+			disableTemplate(templateId);
+		}
+	}, [templateId, templatesByGroup.isSuccess, templatesByGroup.data]);
+
+	const disableTemplate = async (templateId: string) => {
+		try {
+			await updatePreferences.mutateAsync({
+				template_disabled_map: {
+					[templateId]: true,
+				},
+			});
+
+			const allTemplates = Object.values(templatesByGroup.data ?? {}).flat();
+			const template = allTemplates.find((t) => t.id === templateId);
+
+			if (!template) {
+				throw new Error(`Template with ID ${templateId} not found`);
+			}
+
+			displaySuccess(`${template.name} notification has been disabled`);
+
+			queryClient.invalidateQueries(
+				userNotificationPreferences(user.id).queryKey,
+			);
+		} catch (error) {
+			console.error(error);
+			displayError("Error on disabling notification");
```

Comment:
```suggestion
			displayError("An error occurred when attempting to disable the requested notification");
```

Reply:
Is this only catching the `throw` on line 86?
We should also log the error response (if present) of the `mutateAsync` call.

Reply:
this would catch the `mutateAsync` errors as well